docs/java-your-first-app.md

```diff
@@ -0,0 +1,163 @@
+
+<!--SNIPSTART java-project-template-activity-implementation-->
+<!--SNIPEND-->
+
+Save it and run the Workflow and the Worker. The Worker executed the Workflow, and completes the `withdraw()` Activity. But then the exception is thrown when it attempts the `deposit()` Activity. Notice how the Worker keeps retrying the Activity? Visit the UI and click on the RunId of the Workflow. You will see the pending Activity listed there with details such as its state, the number of times it has been attempted, and the next scheduled attempt.
+
+![Activity UI error details](/img/docs/web-ui-activity-error-info.png)
+
+[Timeout configurations](/docs/activities/#timeouts) and [retry policies](/docs/activities/#retries) for Activities are specified in the Workflow code as Activity options. This is another key value proposition that Temporal offers, as other modern applications implement timeout and retry logic within the business process itself. In our Workflow code you can see that we have specified a setStartToCloseTimeout for our Activities, but have not specified a retry policy. That is because Temporal automatically uses a default retry policy if one isn't specified!
+
+So, your Workflow is running, but only the `withdraw()` Activity succeeded. In any other application, the whole process would likely have to be abandoned and rolled back. But with Temporal, we can test the durable state of the Workflow again by stopping either the Temporal server, the Worker, or both and then starting them back up again. The Workflow will be in the same state!
+
+Now, pretend that you have a fix for the issue and you have commented the exception and saved the file. How do we deploy it to a process that is halfway complete? With Temporal, it is actually very simple: just restart the Worker!
+
+On the next scheduled attempt, the Worker will successfully execute the `deposit()` Activity and complete the Workflow. You have just fixed a bug "on the fly" with out losing the state of the Workflow.
+
+## Lore check
+
+![Wisdom](/img/docs/wisdom.png)
+
+Great work! In this tutorial we covered practicalities of running a Temporal Workflow, code implementation, and concepts. Let's do a quick review to make sure you remember some of the more important pieces.
+
+![One](/img/docs/one.png)
+
+What are three of Temporal's value propositions that we touched on in this tutorial?
+
+**Answer**
+
+1. Temporal gives you full visibility in the state of your Workflow and code execution.
+2. Temporal maintains the state of your Workflow, even through server outages and errors.
+3. Temporal makes it easy to timeout and retry Activity code using options that exist outside of your business logic.
+
+![Two](/img/docs/two.png)
+
+How do you pair up Workflow code with a Worker process that executes it?
+
+**Answer**
+
+Use the same Task Queue.
```

Comment:
there's an opportunity here to explain the pros and cons of pairing up the code with the process - why is this not the default, what are the benefits, when would you NOT want to do it, where else in the docs do we learn more?

Reply:
Ya, I agree - though I am thinking this might put us over the "information overload" threshold.
Task routing is such a beast of a concept because it is so flexible, targeting multiple use cases, that I am thinking that just we shouldn't try to address that here, just touch on it - and then once we get another tutorial that specifically covers it - we can link to it from this question.

Reply:
sgtm